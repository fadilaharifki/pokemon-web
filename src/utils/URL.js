export const BASE_URL = 'https://graphql-pokeapi.graphcdn.app/'
export const BASE_URL_ANIMATE = "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/versions/generation-v/black-white/animated"