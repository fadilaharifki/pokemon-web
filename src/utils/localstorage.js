export const createMypokemon = () => {
    if (!localStorage.getItem('_cap_mypokemon')) {
        localStorage.setItem('_cap_mypokemon', JSON.stringify([]))
    }
}