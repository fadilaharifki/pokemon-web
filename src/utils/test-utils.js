import { ApolloProvider } from "@apollo/client";
import { render } from "@testing-library/react";
import { client } from '../graphql/config'
import { BrowserRouter, Routes as ReactRoutes, Route } from "react-router-dom";

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
export const testRendererWithRoute = (ui, { route, path }) => {
  console.log(path);
  render(
    <ApolloProvider client={client}>
      <ReactRoutes>
        <Route path={path} element={ui} />
      </ReactRoutes>
    </ApolloProvider>,
    {
      wrapper: BrowserRouter,
    }
  );
};
