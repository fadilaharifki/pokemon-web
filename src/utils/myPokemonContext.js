import { createContext, useContext, useEffect, useState } from "react";
import { createMypokemon } from "./localstorage";

export const MyPokemonContext = createContext();

export const MyPokemonProvider = (props) => {

    useEffect(() => {
        createMypokemon();
    }, []);

    return (
        <MyPokemonContext.Provider
            value={""}
            {...props}
        />
    );
}

export const useMyPokemon = () => {
    return useContext(MyPokemonContext);
};