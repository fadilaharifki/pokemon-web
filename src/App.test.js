import Footer from "./components/footer";
import Home from './page/home'
import About from './page/about'
import MyPokemon from './page/myPokemont'
import { testRendererWithRoute } from "./utils/test-utils";
import { screen, render, queryByAttribute } from '@testing-library/react'

describe("Footer menu", () => {
  it("render all component", async () => {
    testRendererWithRoute(<Footer />, {
      route: "/",
      path: "/",
    });
  });
});

describe("Find Pokemon", () => {
  it("render page find pokemon", async () => {
    testRendererWithRoute(<Home />, {
      route: "/",
      path: "/",
    });
  });
});

describe("About", () => {
  it("render page about", async () => {
    testRendererWithRoute(<About />, {
      route: "/about",
      path: "/about",
    });
  });

});

describe("Testing tag page about", () => {
  it("render page about", async () => {
    render(<About />);

    const title = screen.getByRole('title');
    const desc = screen.getByRole('desc');
    expect(title).toMatchSnapshot();
    expect(desc).toMatchSnapshot();
  });
})

describe("My Pokemon", () => {
  it("render page My Pokemon", async () => {
    testRendererWithRoute(<MyPokemon />, {
      route: "/mypokemon",
      path: "/mypokemon",
    });
  });
});