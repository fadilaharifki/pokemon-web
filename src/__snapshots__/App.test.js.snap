// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing tag page about render page about 1`] = `
<div
  class="css-bqaej9"
  role="title"
>
  Pokemon APP
</div>
`;

exports[`Testing tag page about render page about 2`] = `
<div
  role="desc"
>
  A simple web application developed with react js, javascript made by 
  <a
    href="https://github.com/fadilaharifki/pokemon-web"
  >
    fadil
  </a>
</div>
`;

exports[`Testing tag page my pokemon render page my pokemon 1`] = `
<div
  class="css-bqaej9"
  role="title"
>
  Pokemon APP
</div>
`;

exports[`Testing tag page my pokemon render page my pokemon 2`] = `
<div
  role="desc"
>
  A simple web application developed with react js, javascript made by 
  <a
    href="https://github.com/fadilaharifki/pokemon-web"
  >
    fadil
  </a>
</div>
`;
